controllers/admin/accessToken.go

```diff
@@ -0,0 +1,88 @@
+package admin
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/owncast/owncast/controllers"
+	"github.com/owncast/owncast/core/data"
+	"github.com/owncast/owncast/models"
+	"github.com/owncast/owncast/utils"
+)
+
+type deleteTokenRequest struct {
+	Token string `json:"token"`
+}
+
+type createTokenRequest struct {
+	Name   string   `json:"name"`
+	Scopes []string `json:"scopes"`
+}
+
+// CreateAccessToken will generate a 3rd party access token.
+func CreateAccessToken(w http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+	var request createTokenRequest
+	if err := decoder.Decode(&request); err != nil {
+		controllers.BadRequestHandler(w, err)
+		return
+	}
+
+	// Verify all the scopes provided are valid
+	if !models.HasValidScopes(request.Scopes) {
+		controllers.BadRequestHandler(w, errors.New("one or more invalid scopes provided"))
+		return
+	}
+
+	token, err := utils.GenerateAccessToken()
+	if err != nil {
+		controllers.InternalErrorHandler(w, err)
+		return
+	}
+
+	if err := data.InsertToken(token, request.Name, request.Scopes); err != nil {
+		controllers.InternalErrorHandler(w, err)
+		return
+	}
+
+	controllers.WriteSimpleResponse(w, true, "created "+request.Name)
```

Comment:
I wonder if returning name and token in json  isn’t more useful long term here?   That way if someone wanted to programmatically or via api create a token

Reply:
Done.